.github/workflows/prebuild.js

```diff
@@ -17,10 +17,12 @@ fs.readFile(footerPath, 'utf8', (err, data) => {
 
     // Replace the NODE_ENV check with the new string
     const updatedData = data.replace(
-        /{process.env.NODE_ENV === "development" \? "Development Mode" : "Production Build"}/,
-        `${(commitLink && shortSHA) ? `Deployed from <a href="${commitLink}">${shortSHA}</a>` : 'Deployed via GitHub Pages'}`
+        '{t(process.env.NODE_ENV === "development" ? "footer.devmode" : "footer.prodmode")}',
+        `${(commitLink && shortSHA) ? `<Trans i18nKey="footer.deployedSHA" values={{ sha: "${shortSHA}" }}><a href="${commitLink}"></a></Trans>` : "{t('footer.deployed')}"}`
```

Comment:
this script is designed to run only on the github actions runner when a build is triggered. these variables are set by github, and the only case when they are set manually are when the script is being run by a user. however, this will not happen naturally.